### ISSUE TITLE
GeoInterface cleanup

### ISSUE THREAD

**Issue by rafaqz** [JuliaGeo/LibGEOS.jl#194]
I noticed a bunch of weird stuff in the geo_interface.jl file during other PRs, like:
https://github.com/JuliaGeo/LibGEOS.jl/blob/1c12df5f0677f2efd1def7b4ecb8ebd9bcaad40f/src/geo_interface.jl#L59
https://github.com/JuliaGeo/LibGEOS.jl/blob/1c12df5f0677f2efd1def7b4ecb8ebd9bcaad40f/src/geo_interface.jl#L49
https://github.com/JuliaGeo/LibGEOS.jl/blob/1c12df5f0677f2efd1def7b4ecb8ebd9bcaad40f/src/geo_interface.jl#L62

It could also do with some organization and comments

**Comment by evetion** [comment]
Agreed. The zero seems like a wrong copy. The other two might be required for Aqua? Good to test though.

**Comment by rafaqz** [comment]
I wondered that too, they at least need "ambiguity" comments if so
